File: backend/schemas/video.py
"""
视频相关 Pydantic 数据模型
"""
from pydantic import BaseModel, Field


class VideoInfoRequest(BaseModel):
    """视频信息查询请求"""
    path: str = Field(..., description='视频文件路径')


class VideoInfoData(BaseModel):
    """视频元数据（与 core.ffmpeg_utils.VideoInfo 对应）"""
    width: int
    height: int
    duration: float
    has_audio: bool
    has_alpha: bool


class VideoInfoResponse(BaseModel):
    """视频信息查询响应"""
    success: bool
    data: VideoInfoData | None = None


class VideoValidateRequest(BaseModel):
    """视频验证请求"""
    path: str = Field(..., description='视频文件路径')


class VideoValidateResponse(BaseModel):
    """视频验证响应"""
    valid: bool
    error: str | None = None


class ExtractFrameRequest(BaseModel):
    """帧提取请求"""
    path: str = Field(..., description='视频文件路径')
    time: float = Field(0.0, description='提取时间点（秒）')
    output_format: str = Field('jpg', description='输出格式（jpg/png）')


class ExtractFrameResponse(BaseModel):
    """帧提取响应"""
    success: bool
    frame_id: str | None = None
    frame_url: str | None = None
    frame_path: str | None = None


class BatchInfoRequest(BaseModel):
    """批量信息查询请求"""
    paths: list[str] = Field(..., description='视频文件路径列表')


class BatchInfoItem(BaseModel):
    """批量查询单项结果"""
    path: str
    success: bool
    error: str | None = None
    info: VideoInfoData | None = None


class BatchInfoResponse(BaseModel):
    """批量信息查询响应"""
    items: list[BatchInfoItem]

File: src/api/video.ts
/**
 * 视频 API 封装
 */
import { get, post } from './client';
import type { VideoInfoData } from '@/types';

/** 获取视频元数据 */
export async function fetchVideoInfo(path: string): Promise<VideoInfoData> {
  const res = await post<{ success: boolean; data: VideoInfoData }>('/api/video/info', { path });
  if (!res.success || !res.data) {
    throw new Error('获取视频信息失败');
  }
  return res.data;
}

/** 验证视频文件 */
export async function validateVideo(path: string): Promise<{ valid: boolean; error?: string }> {
  return post('/api/video/validate', { path });
}

/** 提取视频帧 */
export async function extractFrame(
  path: string,
  time: number = 0,
  format: string = 'jpg'
): Promise<{ frameUrl: string; frameId: string }> {
  const res = await post<{ success: boolean; frame_url: string; frame_id: string }>(
    '/api/video/extract-frame',
    { path, time, output_format: format }
  );
  if (!res.success) {
    throw new Error('帧提取失败');
  }
  return { frameUrl: res.frame_url, frameId: res.frame_id };
}

/** 批量获取视频信息 */
export async function batchFetchVideoInfo(
  paths: string[]
): Promise<Array<{ path: string; success: boolean; info: VideoInfoData | null; error?: string }>> {
  const res = await post<{ items: Array<any> }>('/api/video/batch-info', { paths });
  return res.items;
}
